src/features/machines/swapIntentMachine.ts

```diff
@@ -501,14 +502,37 @@ export const swapIntentMachine = setup({
           ],
         },
 
-        onDone: {
-          target: "Completed",
-
-          actions: {
-            type: "setIntentHash",
-            params: ({ event }) => event.output,
+        onDone: [
+          {
+            target: "Completed",
+            guard: {
+              type: "isOk",
+              params: ({ event }) => event.output,
+            },
+            actions: {
+              type: "setIntentHash",
+              params: ({ event }) => event.output.value as string,
+            },
           },
-        },
+          {
+            target: "Generic Error",
+            actions: [
+              {
+                type: "logError",
+                params: ({ event }) => ({
+                  error: extractPublishIntentError(event.output),
+                }),
+              },
+              {
+                type: "setError",
+                params: ({ event }) => ({
+                  reason: "ERR_CANNOT_PUBLISH_INTENT",
+                  error: extractPublishIntentError(event.output),
```

Comment:
You can actually skip setting error object as it does not provide valuable information. You always attach error specific data to this object:
```ts
// context
 error: null | {
    tag: "err"
    value: 
      {
        reason:
          | "ERR_USER_DIDNT_SIGN"
          // rest...
      } | { 
           reason: "ERR_CANNOT_PUBLISH_INTENT",
           server_reason: string 
      }
  }

// this action
                type: "setError",
                params: ({ event }) => ({
                  reason: "ERR_CANNOT_PUBLISH_INTENT",
                  server_reason: event.output.value,
                })
```

So you can get rid of `extractPublishIntentError` entirely.